packages/widget/src/components/transfer/fungible/fungible-token-transfer.ts

```diff
@@ -76,6 +76,19 @@ export class FungibleTokenTransfer extends BaseComponent {
     }
   };
 
+  private sourceAndDestinationNetworkHasEqualType(): boolean {
+    return (
+      this.transferController.sourceNetwork?.type ===
+      this.transferController.destinationNetwork?.type
```

Comment:
You should check if at least one network is defined otherwise when starting new transfer they are both undefined and wallet address get's set.

Truth be told, there is a bug there because when starting new transfer we shouldn't reset source network